README.md

```diff
@@ -370,6 +374,42 @@ end
 
 It is particularly useful when writing tests.
 
+
+Design Details
+--------------
+
+#### `Mercury#republish`
+
+This method publishes a copy of the message to the _back_ of the
+queue, then acks the original message. This is a similar operation to
+ack/nack/reject. It is only applicable to messages received by
+workers, since listener messages cannot be acknowledged. Unlike other
+acknowledgements, `#republish` takes a continuation (due to the
+publish operation), so the method is located on `Mercury` rather than
+`Mercury::ReceivedMessage`.
```

Comment:
Why is that critical? As long as publisher confirms are enabled, which they are by default (and rabbit_stew enables explicitly, iirc), the message is guaranteed not to be lost (but possibly duplicated, which all our services should be designed to tolerate), so it's no different from a `StandardError` being raised.